build/utils.go

```diff
@@ -65,7 +65,10 @@ func toBuildkitExtraHosts(ctx context.Context, inp []string, nodeDriver *driver.
 	}
 	hosts := make([]string, 0, len(inp))
 	for _, h := range inp {
-		host, ip, ok := strings.Cut(h, ":")
+		host, ip, ok := strings.Cut(h, "=")
+		if !ok {
+			host, ip, ok = strings.Cut(h, ":")
+		}
```

Comment:
Should we update the reference docs in https://github.com/docker/buildx/blob/cb378866587090995fba030bc75a9ca2fb4d8d26/commands/build.go#L489?

Guess it should be `... (format: "host=ip")` if this format is the preferred one now.